lib/cosmos/packet_logs/packet_log_writer.rb

```diff
@@ -246,11 +247,9 @@ def write_packet(packet, take_mutex = true)
           start_new_file()
         end
         if @file
-          entry_header = build_entry_header(packet)
-          if entry_header
-            @file.write(entry_header)
-            @file_size += entry_header.length
-          end
+          build_entry_header(packet) # populate @entry_header
```

Comment:
Before build_entry_header could return nil for no header.  Not it has to return an empty string.   Don't like.